Search edits

ez-may opened the pull request:
Changes made to be compatible with the Recipe entity refactor.

Commit:
Merge of master branch.

Commit:
Updated recipe constructor to reflect published attribute change

Commit:
Merge branch 'recipePage' of github.com:googleinterns/step136-2020 into searchEdits

Commit:
Updated files to be compatible with changes made to the recipe class.

Commit:
Merge branch 'master' of github.com:googleinterns/step136-2020 into searchEdits

Commit:
Merge branch 'master' of github.com:googleinterns/step136-2020 into searchRefactor

Commit:
Fixed master conflict

Commit:
Merge branch 'searchRefactor' into searchEdits

Commit:
Updated composite filter to actually only query public recipes.

Commit:
Made other edits to cope with changes made to the recipe refactor and merge conflicts.

isamitre reviewed (approved):
LGTM

AlexanderEllis commented on src/main/java/com/google/sps/util/TestUtil.java:
```diff
@@ -50,8 +51,8 @@ public static Entity createRecipeEntity(String entityName, String name, String a
         entityName, recipeName, author,
-        message + String.format(" I AM RECIPE %d OF MY KIND.", i + offset),
-        tags, ingredients, steps, Integer.valueOf(0));
+        message + String.format(". I AM RECIPE %d OF MY KIND.", i + offset),
```
What is this additional period at the beginning? Slash what is `message` here?

ez-may replied:
The message is meant to be a string saying what test the recipe is from. For example, in the test files I currently have, it says "THIS IS A TEST RECIPE FOR THE SEARCH BY NAME TEST." This goes in the recipe description box and makes it easy to see whether or not the correct recipe data is being used in a search test. The period helps with formatting, since the string from where it's called had no punctuation. But I think I will move it to the actual method, so that formatting in the future makes more sense.

AlexanderEllis reviewed (approved):
LGTM (one minor comment)

ez-may committed:
Update TestUtil.java

Removed extra period and moving it to the actual message to make formatting clear

ez-may committed:
Update TestData.java

Added period for formatting here.
